src/sds.c

```diff
@@ -603,17 +603,6 @@ int sdsull2str(char *s, unsigned long long v) {
     return l;
 }
 
-/* Create an sds string from a long long value. It is much faster than:
- *
- * sdscatprintf(sdsempty(),"%lld\n", value);
- */
-sds sdsfromlonglong(long long value) {
-    char buf[SDS_LLSTR_SIZE];
-    int len = sdsll2str(buf,value);
-
-    return sdsnewlen(buf,len);
-}
```

Comment:
I want to count the places where sdscatfmt uses "%U, %I, %i, %u" to determine whether it is really necessary to optimize the performance,  I took a rough look and it seems that `sdscatfmt` is just to avoid the inefficiency of stdio's printf, maybe sdsll2str is not the hot path in sdscatfmt :-)

Reply:
not sure i follow you.. it might not be the hot path, but it is use (INFO, CLIENT LIST, and others), if we have a faster version of sdsll2str, why not use it in all places, and avoid the risk of anyone using the slow one by mistake?

i suppose the only reason is to avoid having sds.c depend on util.c, but IMHO it's no longer a concern in redis nowadays.

Reply:
Ok, l'll follow you opinion. let me just do it.

Reply:
> i suppose the only reason is to avoid having sds.c depend on util.c, but IMHO it's no longer a concern in redis nowadays

I would like to explain the my reasons why I don't like util.c: 1) Some functions in util.c used the APIs in sds.c, and it seems weird that sds.c and util.c depend on each other; 2) I think util.c is mess, the name `util` will naturally cause this file to contain various functions.

Reply:
there's generally no problem in C for two c files to depend on each other (server.c, replication.c), the problem here is that we consider sds.c a basic building block.
@yossigo please advise.

Reply:
@oranagra I agree with you we currently consider `sds.c` as part of the code base, so this dependency is not a problem.

Reply:
@oranagra I agree with you we currently consider `sds.c` as part of the code base, so this dependency is not a problem.